pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1265,6 +1265,10 @@ func buildSidecarInboundMgmtListeners(node *model.Proxy, env *model.Environment,
 
 	if managementIP == "" {
 		managementIP = "127.0.0.1"
+		addr := net.ParseIP(node.IPAddresses[0])
+		if addr != nil && addr.To4() == nil {
+			managementIP = "::1"
+		}
```

Comment:
this is a hack.

Reply:
You mean .To4() ? I think it is the standard way to find if an address is ipv6 AFAIK.

Reply:
The logic here is net.ParseIP would return non nil if address is either valid ipv4 or ipv6. Then is address is not nil but To4() returns nil, it means the address is ipv6 address. Please let me know if I misunderstood your comment.